test/unit/toolchain/toolchain_test.bzl

```diff
@@ -111,6 +155,21 @@ def _toolchain_test():
         target_json = ":target_json",
     )
 
+    rust_toolchain(
+        name = "rust_extra_flags_toolchain",
+        binary_ext = "",
+        dylib_ext = ".so",
+        exec_triple = "x86_64-unknown-none",
+        os = "linux",
+        rust_doc = ":mock_rustdoc",
+        rust_std = ":std_libs",
+        rustc = ":mock_rustc",
+        staticlib_ext = ".a",
+        stdlib_linkflags = [],
+        extra_rustc_flags = [TOOLCHAIN_FLAG],
+        target_json = ":target_json",
+    )
+
```

Comment:
Thanks for the tip! I've tried to make that wrapper rule that invokes the transition, but i'm running into an issue where the transition impl errors out because the toolchain hasn't been defined yet:

```
$ bazel test --test_output=streamed //test/toolchain:toolchain_adds_rustc_flags_test
...
ERROR: /mnt/rules_rust/test/toolchain/BUILD.bazel:3:21: While resolving toolchains for target //test/toolchain:lib: invalid registered toolchain ':rust_extra_flags_toolchain': no such target '//:rust_extra_flags_toolchain': target 'rust_extra_flags_toolchain' not declared in package '' defined by /mnt/rules_rust/BUILD.bazel
```

I'm not sure how to change that ordering, given that the `rust_toolchain` is already defined before the rule that uses it. Also not sure how i'd pass which toolchain i wanted to transition to to avoid hardcoding it in the transition impl, but that's only and issue if we wanted to add more tests without duplicating these definitions.

Reply:
Try this in `_transition_impl`:
```
return {"//command_line_option:extra_toolchains": ["@rules_rust//test/toolchain:extra_flags_toolchain"]}
```

My understanding is that this function is evaluated from something like a "top-level" context and requires the labels need to be sufficiently fully qualified.

Reply:
ah thanks that was it